runtime/testy/studentAPI.py

```diff
@@ -7,6 +11,40 @@ def __init__(self, toManager, fromManager):
     self.fromManager = fromManager
     self.toManager = toManager
     self._createSensorMapping()
+    self._coroutines_running = set()
+
+  def run(self, fn, *args, **kwargs):
+    """
+    Starts a "coroutine", i.e. a series of actions that proceed 
+    independently of the main loop of code.
+
+    The first argument must be a function defined with 'async def'.
+
+    The remaining arguments are then passed to that function before it is
+    called.
+
+    Multiple simultaneous coroutines that use the same robot actuators will
+    lead to surprising behavior. To help guard against errors, calling
+    `coroutine` with a `fn` argument that is currently running is a no-op.
+    """
+
+    if not inspect.isfunction(fn):
+        raise ValueError("First argument to Robot.coroutine must be a function")
+    elif not inspect.iscoroutinefunction(fn):
+        raise ValueError("First argument to Robot.coroutine must be defined with `async def`, not `def`")
+
+    if fn in self._coroutines_running:
+        return
+
+    self._coroutines_running.add(fn)
+
+    future = fn(*args, **kwargs)
```

Comment:
I'm not sure this can be moved. Calling the function here means that any errors involved in calling the function (such as invalid number of arguments) get caught immediately and print decent stack traces, as opposed to getting caught later inside `wrapped_future`, where the stack trace would point into the event loop.